internal/controller/controllers/local_cluster_import_controller.go

```diff
@@ -0,0 +1,528 @@
+}
+
+func (r *LocalClusterImportReconciler) deleteClusterDeployment(ctx context.Context, namespace string, name string) error {
+	clusterDeployment := &hivev1.ClusterDeployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+	}
+	if err := r.client.Delete(ctx, clusterDeployment); err != nil && !k8serrors.IsNotFound(err) {
+		return errors.Wrapf(err, "failed to delete ClusterDeployment %s in namespace %s", name, namespace)
+	}
+	return nil
+}
+
+func (r *LocalClusterImportReconciler) deleteAgentClusterInstall(ctx context.Context, namespace string, name string) error {
+	agentClusterInstall := &hiveext.AgentClusterInstall{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+	}
+	if err := r.client.Delete(ctx, agentClusterInstall); err != nil && !k8serrors.IsNotFound(err) {
+		return errors.Wrapf(err, "failed to delete AgentClusterInstall %s in namespace %s", name, namespace)
+	}
+	return nil
+}
+
+func (r *LocalClusterImportReconciler) hasLocalManagedCluster(ctx context.Context) (bool, error) {
+	managedCluster := &clusterv1.ManagedCluster{}
+	namespacedName := types.NamespacedName{
+		Name: r.localClusterName,
+	}
+	err := r.client.Get(ctx, namespacedName, managedCluster)
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
+			return false, nil
+		}
+		return false, err
+	}
+	return true, nil
+}
+
+// This will ensure that the ClusterDeployment and AgentClusterInstall are not present
+// If they are present, they will be deleted
+// No error will be returned if these are not found as this is the desired state.
+func (r *LocalClusterImportReconciler) ensureLocalClusterCRsDeleted(ctx context.Context) error {
+	err := r.deleteClusterDeployment(ctx, r.localClusterName, r.localClusterName)
+	if err != nil && !k8serrors.IsNotFound(err) {
+		r.log.Errorf("could not delete local ClusterDeployment due to error %s", err.Error())
+		return err
+	}
+	err = r.deleteAgentClusterInstall(ctx, r.localClusterName, r.localClusterName)
+	if err != nil && !k8serrors.IsNotFound(err) {
+		r.log.Errorf("could not delete local AgentClusterInstall due to error %s", err.Error())
+		return err
+	}
+	return nil
+}
+
+func (r *LocalClusterImportReconciler) createOrUpdateClusterImageSet(ctx context.Context, release_image string, instance *aiv1beta1.AgentServiceConfig) error {
+	clusterImageSet := hivev1.ClusterImageSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: localClusterImageSetName,
+		},
+	}
+	mutateFn := func() error {
+		if err := controllerutil.SetControllerReference(instance, &clusterImageSet, r.client.Scheme()); err != nil {
+			return err
+		}
+		clusterImageSet.Spec.ReleaseImage = release_image
+		return nil
+	}
+	createOrUpdateResult, err := controllerutil.CreateOrUpdate(ctx, r.client, &clusterImageSet, mutateFn)
+	if err != nil {
+		return errors.Wrap(err, "could not create cluster image set")
+	}
+	r.log.Infof("%s ClusterImageSet %s", createOrUpdateResult, clusterImageSet.Name)
```

Comment:
I think we shouldn't log this if the result is `OperationResultNone`

Note the possible results in https://github.com/kubernetes-sigs/controller-runtime/blob/cf69be219f050a456a93949d0c53773836f9216c/pkg/controller/controllerutil/controllerutil.go#L253-L264

Also looking at the values there I think the log would read better as is suggested in the link `<resources> <namespace>/<name> has been <result>`